src/utils/DragHandler.ts

```diff
@@ -66,77 +65,55 @@ class DragHandler {
-          param: {
-            elementId: id,
-            x: this.dx,
-            y: -this.dy,
-          }
-        };
-
-        paramArray.push(dragAction);
-
-        if (el.classList.contains('divLine') || el.classList.contains('accid') || el.classList.contains('custos')) {
-          const { clientX, clientY } = d3.event.sourceEvent;
-          const newStaff = getStaffIdByCoords(clientX, clientY);
+    // If the selection has not been moved more than 5 pixels in any direction,
+    // do not do any action
+    const xDiff = Math.abs(this.dx), yDiff = Math.abs(this.dy);
+    if (xDiff <= 5 && yDiff <= 5) {
+      this.reset();
+      this.dragInit();
+      return;
+    }
 
-          const staffAction: ChangeStaffToAction = {
-            action: 'changeStaffTo',
-            param: {
-              elementId: id,
-              // if divline is moved to the background (and not a staff),
-              // set the staffId to the original staff
-              staffId: newStaff ? newStaff : el.closest('.staff').id,
-            }
-          };
+    // Filter selection for draggable elements
+    const selection = this.selection.filter((el) => !el.classList.contains('resizePoint'));
 
-          paramArray.push(staffAction);
-        }
-      });
+    // If the cursor is out of bounds or the selection array is out of bounds,
+    // drag actions should not happen. Return an error notification and reset
+    // the drag handler
+    if (this.isCursorOutOfBounds() || this.isSelOutOfBounds(selection)) {
+      this.returnElements(this.selection);
+      this.reset();
+      this.dragInit();
+      return queueNotification('[FAIL] Glyphs were placed out of bounds! Drag action failed.');
+    }
 
+    // Create the chain editor action for selection
+    const dragActions = this.createDragActions(selection);
     const editorAction: EditorAction = {
       action: 'chain',
-      param: paramArray
+      param: dragActions
     };
 
-    const xDiff = Math.abs(this.dx);
-    const yDiff = Math.abs(this.dy);
-
-    if (xDiff > 5 || yDiff > 5) {
-      this.neonView
-        .edit(editorAction, this.neonView.view.getCurrentPageURI())
-        .then(() => {
-          this.neonView.updateForCurrentPage();
-          this.endOptionsSelection();
-          this.reset();
-          this.dragInit();
-        });
-    } else {
-      this.reset();
-      this.dragInit();
-    }
+    // Send editor action
+    this.neonView
+      .edit(editorAction, this.neonView.view.getCurrentPageURI())
+      .then(() => {
+        this.neonView.updateForCurrentPage();
+        this.endOptionsSelection();
+        this.reset();
+        this.dragInit();
+      });
   }
 
   /** Set the d3 action to use for [[reset]]. */
-  resetTo (reset: (selection: d3.Selection<d3.BaseType, {}, HTMLElement, any>, args: any[]) => void): void {
+  resetTo (reset: (selection: d3.Selection<d3.BaseType, unknown, HTMLElement, unknown>, args: unknown[]) => void): void {
```

Comment:
I had 3 warnings from ESlint every time I opened this file, so I finally just changed it to 'unknown'